Add Github Actions test workflow

wrongkindofdoctor opened the pull request:
**Description**
Add a Github Actions workflow file to run the MDTF_diagnostics unit tests and a subset of PODs using synthetic observational data in macOS and Linux environments. The CI is  triggered by pushes to the main and develop branches. 


Associated issue #171 

**How Has This Been Tested?**
PODs and unit tests are run using the macos-latest and ubuntu-latest Github Actions environments. 

**Checklist:**
- [x] I have reviewed my own code to ensure that if follows the [POD development guidelines](https://mdtf-diagnostics.readthedocs.io/en/latest/sphinx/dev_guidelines.html)
- [ ] The script are written in Python 3.6 or above (preferred; required if funded by a CPO grant), NCL, or R
- [ ] All of my scripts are in the diagnostics/ subdirectory, and include a main_driver script template html, and settings.jsonc
- [ ] The main_driver script header has all of the information in the POD documentation, excluding the "More about this diagnostic" section
- [ ] The POD directory and html template have the same short name as my POD
- [ ] The html template has a 1-paragraph synopsis of the POD and links to the main documentation
- [x] If applicable, I've added a .yml file to src/conda, and my environment builds with conda_env_setup.sh 
- [ ] The POD scripts do not access the internet or networked resources
- [x] I have commented the code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation in the POD's doc/ subdirectory
- [ ] I have created the directory input_data/obs_data/[pod short name]
- [x] My code is portable; it uses [MDTF environment variables](https://mdtf-diagnostics.readthedocs.io/en/latest/sphinx/ref_envvars.html), and does not contain hard-coded file or directory paths
- [ ] I have added information about the raw data files to the documentation
- [ ] I have provided the code to generate digested data files from raw data files
- [ ] Each digested data file generated by the script contains numerical data (no figures), and is 3 GB or less in size
- [x] The repository contains no extra test scripts or data files
- [x] My branch is up-to-date with the NOAA-GFDL develop branch, and all merge conflicts are resolved

jkrasting commented on .github/workflows/mdtf_tests.yaml:
```diff
@@ -0,0 +1,162 @@
+        # attempt FTP data fetch
+        # allow 20 min for transfer before timeout; Github actions allows 6 hours for individual
+        # jobs, but we don't want to max out resources that are shared by the NOAA-GFDL repos.
+        #curl --verbose --ipv4 --connect-timeout 8 --max-time 1200 --retry 128 --ftp-ssl --disable-epsv --ftp-pasv -u "anonymous:anonymous" ftp://ftp.gfdl.noaa.gov/pub/jessica.liptak/example_obs_data.tar --output example_obs_data.tar
```
@wrongkindofdoctor - we should request some permanent ftp space that won't be subject to the sweeper.

wrongkindofdoctor replied:
Agreed. I can submit a helpdesk ticket for permanent space under the role account.

tsjackson-noaa replied:
Good catch -- I wasn't aware that files in users' public FTP directories get swept.

wrongkindofdoctor replied:
The yaml has been updated to point to the obs_data files in the permanent FTP directory

jkrasting commented on src/conda/env_synthetic_data.yml:
```diff
@@ -0,0 +1,17 @@
+# Conda environment for synthetic data generation
```
@wrongkindofdoctor - in your testing, do you feel that the MDTF synthetic data package is mature enough that it should go up on PyPi and anaconda.org to make installation easier.  I can do this if you think it's ready and would be helpful.

wrongkindofdoctor replied:
I think that that the synthetic data package should be modified to be more flexible. It relies on hard-coded variables and metadata used by some of the current PODs, but this will be difficult to maintain as we add more PODs, and support for CMIP-style variables, to the CI. If we add an arparse routine to accept GFDL, CESM, or CMIP as the convention, consolidate the GFDL and NCAR synthetic scripts into one main script, and move the variables and metadata to jsons (or some other format), we can extend the variables by amending/adding spec files.

jkrasting reviewed (commented):
Looks great, @wrongkindofdoctor.  I made some comments but none of them are critical enough that they need to be fixed with this PR.

wrongkindofdoctor commented:
@jkrasting @tsjackson-noaa All right, the yaml file should be good to go after updating the FTP paths and rebasing.

jkrasting reviewed (dismissed):
Looks good, thanks @wrongkindofdoctor!

tsjackson-noaa reviewed (commented):
Hi @wrongkindofdoctor --
Thanks for all the work you did setting up GH actions -- one minor point:

As currently written, the `_env_synthetic_data` conda environment will be picked up by the [shell script](https://github.com/NOAA-GFDL/MDTF-diagnostics/blob/main/src/conda/conda_env_setup.sh) for user installation of conda environments. I'd request:
1) If synthetic data is a CI-only thing (at least for the moment; I think we should write more docs before rolling it out to users), rename the file to `_env_synthetic_data.yml` so it won't be picked up by the glob in that script. (I should do the same for `env_dev.yml`).
2) The name given in the YAML file should be changed to something like `_MDTF_synthetic_data`, to go with the convention used by all the other envs (to make it clear to the user where they all came from.)

Sorry it took me so long to get around to reviewing -- if you address the issue above I promise I'll reapprove on the same day!

wrongkindofdoctor commented:
@tsjackson-noaa  All right, I changed the file name and added a line to install the test environment separately from the POD environments. I also cleaned up some of the test files that were retained in the initial PR.
